build.gradle.kts

```diff
@@ -58,26 +58,25 @@ kotlin {
   jvmToolchain(21)
 }
 
+repositories {
+  mavenCentral()
+}
+
 tasks {
   withType<KotlinCompile> {
     compilerOptions.jvmTarget = JVM_21
     compilerOptions.freeCompilerArgs = listOf("-Xjvm-default=all")
+  }
+}
+
+tasks.test {
+  finalizedBy(tasks.jacocoTestReport)
+}
 
-    named<JacocoReport>("jacocoTestReport") {
-      dependsOn(named("check"))
-      reports {
-        html.required.set(true)
-        xml.required.set(true)
-      }
-    }
-    finalizedBy(named("jacocoTestReport"))
-    named("assemble") {
-      doFirst {
-        delete(
-          fileTree(project.layout.buildDirectory.get())
-            .include("libs/*.jar"),
-        )
-      }
-    }
+tasks.jacocoTestReport {
+  dependsOn(tasks.test)
+  reports {
+    html.required.set(true)
+    xml.required.set(true)
```

Comment:
probably need to test coverage works but getting deployment working probably trumps that!